File: Array/45_jump_game_II.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        n, window, jumps, limit = len(nums), 0, 0, 0
        if n == 1:
            return 0

        for i in range(n):
            window = max(window, i + nums[i])
            if i == limit:
                limit = window
                jumps += 1

                if window >= n - 1:
                    return jumps

        return jumps

File: Array/45_jump_game_II.cpp
class Solution {
    public:
        int jump(vector<int>& nums) {
            int jumps = 0, limit = 0, window = 0;
    
            for(int i = 0; i < nums.size() - 1; i += 1) {
                if(i + nums[i] > window) {
                    window = i + nums[i];
                }
    
                if(i == limit) {
                    limit = window;
                    jumps += 1;
                }
            }
    
            return jumps;
        }
    };

// revised - 02/26/2025